### PR TITLE
fix(preheat):handle fail case of preheat in job

### PR DESCRIPTION
**Pull request by steven-zou**
Signed-off-by: Steven Zou <szou@vmware.com>

### CODE REVIEW

File: src/pkg/p2p/preheat/job.go [hunk 1]
```diff
@@ -185,9 +185,15 @@ func (j *Job) Run(ctx job.Context, params job.Parameters) error {
 
 			myLogger.Infof("Check preheat progress: %s", s)
 
-			// Finished
-			if s.Status == provider.PreheatingStatusSuccess {
+			switch s.Status {
+			case provider.PreheatingStatusFail:
+				// Fail
+				return preheatJobRunningError(errors.Errorf("preheat failed: %s", s))
```
Comment (chlins) [inline src/pkg/p2p/preheat/job.go:191]: errors.Errorf("preheat failed: %s", s),  s is a struct? not a string, so maybe need change to  %v or s.Status.

**Review by chlins — APPROVED** [review]
LGTM

File: src/pkg/p2p/preheat/job.go (hunk 1 above)
Comment (steven-zou) [reply src/pkg/p2p/preheat/job.go:191]: s has `String()` method. So it's ok.

```go
func (ps *PreheatingStatus) String() string
```

**Review by kofj — APPROVED** [review]
/LGTM

**Pull request merged**